src/stenotype/cli.py

```diff
@@ -0,0 +1,82 @@
+"""Command Line Interface for stenotype.
+
+The backend is exposed here through a simple command line interface. Documentation of
+their usage is contained in the docstrings, which are written in a way that they will be
+rendered nicely by click's automatic --help text generator.
+"""
+from logging import getLogger
+
+import click
+
+from stenotype.backend import parser
+from stenotype import util
+
+log = getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    "--loglevel",
+    default="INFO",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR"]),
+    help="Set the loglevel.",
+)
+@click.option(
+    "-f",
+    "--file",
+    type=click.File("r"),
+    help="Load a reference file if annotations need custom classes.",
+)
+@click.option(
+    "-s",
+    "--shorten",
+    is_flag=True,
+    help="Inverts the parser to instead try to generate stenotype from standard types.",
+)
+@click.option(
+    "-c",
+    "--check",
+    is_flag=True,
+    help="Disables the check-skipping. Annotations need to be written as they would "
+    "be in source files in this mode.",
+)
+@click.argument("args", nargs=-1)
+def cli(args, loglevel, file, shorten, check):
+    """CLI entrypoint to test the stenotype backend.
+
+    Enter any number of stenotype expressions to see which standard annotations they will
+    resolve into. There is no need to double quote expression or to assign them to a
+    value, the following will just work:
+
+    \b
+      $ stenotype 'bool or int'
+      typing.Union[bool, int]
+      $ stenotype '(int) -> bool'
+      typing.Signature[[int], bool]
+
+    You can also enter multiple arguments that will be executed in turn and consider each
+    other in the context of, for example, custom type variables:
+
+    \b
+      $ stenotype 'T, int' 'bool or T'
+      T = TypeVar[int]
+      typing.Union[bool, T]
+    """
+    util.setup_logging(loglevel)
+
+    # early exits
+    if shorten and check:
+        print("The flags --shorten and --check can not be used at the same time.")
```

Comment:
Consider writing diagnostics to ``stderr`` -- this allows safely invoking ``stenotype`` by other tools to get the translated types.

Reply:
good call